src/App.tsx

```diff
@@ -1,10 +1,44 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { USER_ID, getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+import { Header } from './components/Header/Header';
+import { TodoList } from './components/TodoList';
+import { Error } from './components/Error/Error';
+import { Footer } from './components/Footer/Footer';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState('');
+  const [status, setStatus] = useState(Status.All);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setError('Unable to load todos');
+        setTimeout(() => setError(''), 3000);
+      });
+  }, []);
+
+  function getVisibleToDos(newTodos: Todo[], newStatus: Status) {
```

Comment:
This is a pure function, and it does not check on the state by itself, it needs to receive the state.
If that is the case, you should probably declare this function outside the component so it doesn't get re-declared on every re-render